pkg/segment/query/processor/queryprocessor.go

```diff
@@ -431,3 +436,34 @@ func createEmptyResponse(queryType structs.QueryType) *structs.PipeSearchRespons
 
 	return response
 }
+
+func validateStreamStatsTimeWindow(firstAgg *structs.QueryAggregators) error {
+
+	for curAgg := firstAgg; curAgg != nil; curAgg = curAgg.Next {
+		if curAgg.StreamstatsExpr != nil && curAgg.StreamstatsExpr.TimeWindow != nil {
+			// Check all previous commands up to this point
+			hasSort := false
+			timeSort := false
+			timeSortAsc := true
+
+			// Check commands that come before streamstats
+			for prevAgg := firstAgg; prevAgg != curAgg; prevAgg = prevAgg.Next {
```

Comment:
Why do we want to iterate twice? can't we just start from firstAgg and maintain whether the sort is present or not, if so is it on timestamp.